ci/prow/make_config.go

```diff
@@ -807,33 +991,45 @@ func executeJobTemplateWrapper(repoName string, data interface{}, generateOneJob
 			}
 		}
 	}
-	if len(legacyBranches) == 0 { // Generate only one job as normal if LegacyBranches is not set
-		generateOneJob(data)
-	} else {
-		// Generate one job with 'knative.dev' path alias for branches other than legacy branches,
-		// and another job without path alias for legacy branches
-		var base *baseProwJobTemplateData
-		switch v := data.(type) {
-		case *presubmitJobTemplateData:
-			base = &data.(*presubmitJobTemplateData).Base
-		case *postsubmitJobTemplateData:
-			if strings.HasSuffix(data.(*postsubmitJobTemplateData).PostsubmitJobName, "go-coverage") {
-				generateOneJob(data)
-				return
+	if len(legacyBranches) > 0 {
+		sbs = append(sbs, specialBranchLogic{
+			branches: legacyBranches,
+			opsNew: func(base *baseProwJobTemplateData) {
+				base.PathAlias = "path_alias: knative.dev/" + base.RepoName
+				base.ExtraRefs = append(base.ExtraRefs, "  "+base.PathAlias)
+			},
+			restore: func(base *baseProwJobTemplateData) {
+				base.PathAlias = ""
+				base.ExtraRefs = base.ExtraRefs[:len(base.ExtraRefs)-1]
+			},
+		})
+	}
+
+	var go112Branches []string
+	// Find out if Go112Branches is set in repo settings
+	for _, repo := range repositories {
+		if repo.Name == repoName {
+			if len(repo.Go112Branches) > 0 {
+				go112Branches = repo.Go112Branches
 			}
-			base = &data.(*postsubmitJobTemplateData).Base
-		default:
-			log.Fatalf("Unrecognized job template type: '%v'", v)
 		}
-		branches := base.Branches
-		skipBranches := base.SkipBranches
-		base.PathAlias = ""
-		base.Branches, base.SkipBranches = consolidateBranches(branches, skipBranches, legacyBranches, make([]string, 0))
-		generateOneJob(data)
-		base.Branches, base.SkipBranches = consolidateBranches(branches, skipBranches, make([]string, 0), legacyBranches)
-		base.PathAlias = "path_alias: knative.dev/" + base.RepoName
-		base.ExtraRefs = append(base.ExtraRefs, "  "+base.PathAlias)
+	}
+	if len(go112Branches) > 0 {
+		sbs = append(sbs, specialBranchLogic{
+			branches: go112Branches,
+			opsNew: func(base *baseProwJobTemplateData) {
+				base.Image = getGo113ImageName(base.Image)
```

Comment:
I am thinking how much change we will need if one day we need to update to Go 1.14, and seems it will be a lot.

Do you think it will be a concern for us? If yes I think `go-version: 1.13` in the meta config would be better.

Reply:
It should be much less change than this PR, since this one set the foundation for future diverges like this. But open to discussion